client/client.go

```diff
@@ -0,0 +1,112 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+type (
+	// Error is the error type returned by the API in case the status code
+	// is not a 2xx code.
+	Error struct {
+		Message string `json:"message"`
+	}
+
+	// Client is a helper library for interacting with an API.
+	Client struct {
+		staticAddr string
+	}
+)
+
+// Error implements the error interface for the Error type. It returns only the
+// Message field.
+func (err Error) Error() string {
+	return err.Message
+}
+
+// NewClient creates a new Client for an API listening on the given address.
+func NewClient(addr string) *Client {
+	return &Client{
+		staticAddr: addr,
+	}
+}
+
+// readAPIError decodes and returns an api.Error.
+func readAPIError(r io.Reader) error {
+	var apiErr Error
+	b, _ := ioutil.ReadAll(r)
+	if err := json.NewDecoder(bytes.NewReader(b)).Decode(&apiErr); err != nil {
+		return errors.AddContext(err, "could not read error response")
+	}
+	return apiErr
+}
+
+// get performs a GET request on the provided resource.
+func (c *Client) get(resource string, headers map[string]string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", c.staticAddr+resource, nil)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	return http.DefaultClient.Do(req)
+}
+
+// post performs a POST request on the provided resource.
+func (c *Client) post(resource string, headers map[string]string, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequest("POST", c.staticAddr+resource, body)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	return http.DefaultClient.Do(req)
+}
```

Comment:
nit: Can we DRY this up? It's not a big deal but... it's like a global variable called `tmp` - it just irks me. :)